configure.ac

```diff
@@ -1608,6 +1608,24 @@ AC_CHECK_MEMBER([struct lyd_node.priv], [], [
   ])
 ], [[#include <libyang/libyang.h>]])
 
+ac_ld_flag_save="$LDFLAGS"
+LDFLAGS="$LDFLAGS $libyang_LIBS"
```

Comment:
Just a suggestion, you might want to set CPPFLAGS too, since $libyang_CFLAGS can point to a custom include path.

Reply:
note that the current check for libyang.h doesn't do this either, I was planning to fix that.

Reply:
`AC_CHECK_FUNC` won't include `libyang.h` anyway, it just checks for the existence of the function at the linker level regardless of its prototype.

However, we are indeed not using `libyang_CFLAGS` nor `libyang_LIBS` anywhere (`lib/subdir.am` has a hard `-lyang` so that needs to be fixed.)

Reply:
> AC_CHECK_FUNC won't include libyang.h anyway, it just checks for the existence of the function at the linker level regardless of its prototype.

Ah yes you're right of course.

Reply:
Added a 2nd commit to go over a bunch of similar issues & fix them.